network/network-address/src/lib.rs

```diff
@@ -321,6 +321,45 @@ impl NetworkAddress {
             .push(Protocol::Handshake(handshake_version))
     }
 
+    /// Check that a `NetworkAddress` looks like a typical LibraNet address with
+    /// associated protocols.
+    pub fn is_libranet_addr(&self) -> bool {
```

Comment:
is it something that potentially we should call all the time when deserializing a NetowkAddress?

Reply:
potentially having a `deserialize_unchecked_addr()`  function to deserialize without checking if this has a real use case (making the default behavior sane)

Reply:
not all addrs include the later handshake protocols, like listen_address

Reply:
why don't they have the handshake protocol?

also, what about having different `from` functions then? a `from_libra_addr`, and so on.

I guess my comment is more about, how can we make this API harder to mis-use :)

Reply:
This is a very good point... but what is a libranet address? :/
Perhaps for listen address we just use a SockAddr? There's no real benefit to using a multiaddr there.

Reply:
isn't `SocketAddr` too restrictive? can't listen on `"/memory/<port>"` anymore